TestExtractTarOverwrite: fails when running as root

I cloned the rkt git repository in `/var/tmp/rkt`, compiled and run the unit test TestExtractTarOverwrite as root. It fails with the following:

```
ok          github.com/coreos/rkt/common        1.012s        coverage: 39.2% of statements
ok          github.com/coreos/rkt/common/cgroup        1.011s        coverage: 8.9% of statements
ok          github.com/coreos/rkt/pkg/fileutil        1.010s        coverage: 35.1% of statements
ok          github.com/coreos/rkt/pkg/keystore        1.291s        coverage: 60.9% of statements
ok          github.com/coreos/rkt/pkg/lock        1.010s        coverage: 77.2% of statements
ok          github.com/coreos/rkt/pkg/set        1.008s        coverage: 100.0% of statements
--- FAIL: TestExtractTarOverwrite (1.02s)
tar_test.go:620: unexpected error: Expected file "folder01/file02" not in files
FAIL
```

/cc @dgonyeo

--- a/pkg/tar/tar_test.go
+++ b/pkg/tar/tar_test.go
@@ -427,7 +427,7 @@ func TestExtractTarOverwrite(t *testing.T) {
 	if !sys.HasChrootCapability() {
 		t.Skipf("chroot capability not available. Disabling test.")
 	}
-	testExtractTarOverwrite(t, extractTarHelper)
+	testExtractTarOverwrite(t, extractTarOverwriteHelper)
 }
 func TestExtractTarOverwriteInsecure(t *testing.T) {
 	testExtractTarOverwrite(t, extractTarInsecureHelper)
@@ -785,6 +785,10 @@ func testExtractTarHardLink(t *testing.T, extractTar func(io.Reader, string) err
 	if origFile.Ino != linkedFile.Ino {
 		t.Errorf("original file and linked file have different inodes")
 	}
+}
+
+func extractTarOverwriteHelper(rdr io.Reader, target string) error {
+	return ExtractTar(rdr, target, true, user.NewBlankUidRange(), nil)
 }
 
 func extractTarHelper(rdr io.Reader, target string) error {

